### ISSUE TITLE
Enforce Password & expiration date on mail sharing

### ISSUE THREAD

**Issue by RumblyShip** [nextcloud/server#11547]
### Steps to reproduce
1. Upload a file
2. Share via email
3. Not forced to password protect

### Expected behaviour
I would like a configuration option to enforce password use like the option that is available for sharing via a link

### Actual behaviour
Able to share without a password

### Server configuration

**Operating system**: Ubuntu 18.04

**Web server:** nginx
**Database:** MariaDB
**PHP version:** 7.2
**Nextcloud version:** 14.0.1
**Updated from an older Nextcloud/ownCloud or fresh install:** fresh install
**Where did you install Nextcloud from:** manual install
**Signing status:**
<details>
<summary>Signing status</summary>
No errors have been found.
</details>

**List of activated apps:**
<details>
<summary>App list</summary>

```
Enabled:
  - activity: 2.7.0
  - admin_audit: 1.4.0
  - bruteforcesettings: 1.1.0
  - cloud_federation_api: 0.0.1
  - dav: 1.6.0
  - federatedfilesharing: 1.4.0
  - files: 1.9.0
  - files_automatedtagging: 1.4.0
  - files_retention: 1.3.0
  - files_sharing: 1.6.2
  - logreader: 2.0.0
  - lookup_server_connector: 1.2.0
  - nextcloud_announcements: 1.3.0
  - notifications: 2.2.1
  - oauth2: 1.2.1
  - password_policy: 1.4.0
  - provisioning_api: 1.4.0
  - serverinfo: 1.4.0
  - sharebymail: 1.4.0
  - systemtags: 1.4.0
  - theming: 1.5.0
  - twofactor_backupcodes: 1.3.1
  - updatenotification: 1.4.1
  - user_ldap: 1.4.0
  - workflowengine: 1.4.0

```
</details>

**Nextcloud configuration:**
<details>
<summary>Config report</summary>

```
{
    "system": {
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "***REMOVED SENSITIVE VALUE***"
        ],
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "dbtype": "mysql",
        "version": "14.0.1.1",
        "overwrite.cli.url": "***REMOVED SENSITIVE VALUE***",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbport": "",
        "dbtableprefix": "oc_",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "htaccess.RewriteBase": "\/",
        "skeletondirectory": "\/NCSkeleton",
        "memcache.local": "\\OC\\Memcache\\Redis",
        "redis": {
            "host": "***REMOVED SENSITIVE VALUE***",
            "port": 6379
        },
        "updater.release.channel": "production",
        "lost_password_link": "mailto:***REMOVED SENSITIVE VALUE***",
        "ldapIgnoreNamingRules": false,
        "ldapProviderFactory": "OCA\\User_LDAP\\LDAPProviderFactory",
        "mail_from_address": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpmode": "smtp",
        "mail_smtpauthtype": "LOGIN",
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpport": "25",
        "mail_smtpsecure": "tls",
        "simpleSignUpLink.shown": false
    }
}

```
</details>

**Are you using external storage, if yes which one:** local

**Are you using encryption:** no

**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory

### Client configuration
**Browser:** Chrome
**Operating system:** Windows 10

**Comment by jancborchardt** [comment]
@RumblyShip this is actually possible:
![nextcloud password protect via email](https://user-images.githubusercontent.com/925062/58111754-5af7c480-7bf2-11e9-8260-86942701abf0.png)

Or do you mean that if "Enforce password protection" under link sharing in the Admin settings is enabled, that it should also be enforced for mail shares?
![sharing enforce setting](https://user-images.githubusercontent.com/925062/58111932-b6c24d80-7bf2-11e9-8099-ef289986b5a1.png)

**Comment by skjnldsv** [comment]
Actually, same goes for email expiration date, not only password

**Comment by RumblyShip** [comment]
I would like to see an option to enforce passwords in email sharing. We have a policy that anything shared externally requires a password to access. 

Last I tested, this only was enforceable in link shares and not on email.

**Comment by skjnldsv** [comment]
Fixed with #15719
